package/src/components/ChannelList/ChannelList.tsx

```diff
@@ -164,6 +166,34 @@ export type ChannelListProps<
     setChannels: React.Dispatch<React.SetStateAction<Channel<StreamChatGenerics>[] | null>>,
     event: Event<StreamChatGenerics>,
   ) => void;
+  /**
+   * Override the default listener/handler for event `message.new`
+   * This event is received on channel, when a new message is added on a channel.
+   *
+   * @param lockChannelOrder If set to true, channels won't dynamically sort by most recent message, defaults to false
+   * @param setChannels Setter for internal state property - `channels`. It's created from useState() hook.
+   * @param event An [Event object](https://getstream.io/chat/docs/event_object) corresponding to `message.new` event
+   *
+   * @overrideType Function
+   * */
+  onNewMessageEvent?: (
+    lockChannelOrder: boolean,
+    setChannels: React.Dispatch<React.SetStateAction<Channel<StreamChatGenerics>[] | null>>,
+    event: Event<StreamChatGenerics>,
+  ) => void;
+  /**
+   * Override the default listener/handler for event `notification.message_new`
+   * This event is received on channel, which is not being watched.
+   *
+   * @param setChannels Setter for internal state property - `channels`. It's created from useState() hook.
+   * @param event An [Event object](https://getstream.io/chat/docs/event_object) corresponding to `notification.message_new` event
+   *
+   * @overrideType Function
+   * */
+  onNewMessageNotification?: (
```

Comment:
in terms of naming convention we should add the Suffix event to it just like `onNewMessageEvent`